### PR TITLE
Update to latest Theia: 1.17.1

### PR DESCRIPTION
**Pull request by marcdumais-work**
#### What it does

Updates Blueprint to use the latest Theia framework release, v1.17.1

#### How to test

Confirm that Blueprint builds and that the generated package(s) still work as expected.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
@marcdumais-work I added a commit on top to resolve the build and runtime issues following the update:
- fix `setMenu` expectation which now accepts `null`.
- fix `es5` to `ES2017` target which caused runtime issues

I also took the liberty of refactoring some of the `inversify` imports to instead use the shared dependency from `@theia/core/shared/`.

File: configs/base.tsconfig.json [hunk 1]
```diff
@@ -14,13 +14,13 @@
...
         "lib": [
-            "es6",
+            "ES2017",
             "dom"
         ],
         "sourceMap": true,
         "composite": true
     }
-}
+}
```
Comment (vince-fugnitto) [inline configs/base.tsconfig.json:26]: I'll amend the fix commit to restore the newline :+1:

Comment (marcdumais-work) [reply configs/base.tsconfig.json:26]: I do not think I even opened that file. Can such an edit happen on its own?

Comment (marcdumais-work) [reply configs/base.tsconfig.json:26]: The fix commit is now on the bottom, so maybe better that I amend my commit?

Comment (vince-fugnitto) [reply configs/base.tsconfig.json:26]: I did in the fix (to update to es2017) :)
- a475368f24dfd05f298ce7e1b1a418f1dfee71ea

**Review by vince-fugnitto — APPROVED** [review]
I confirm that the changes work correctly :+1: 
- [x] the build passes with the updates
- [x] the application successfully starts
- [x] the application successfully packages, and starts as a result
- [x] fixes in `1.17.1` work correctly

**Pull request merged**